src/main/java/com/example/coloringpage/ImageToColoringPageConverter.java

```diff
@@ -77,16 +75,47 @@ private BufferedImage invertImage(BufferedImage image)
         return image;
     }
 
+    /**
+     *
+     * @param image The image to be blurred
+     * @return A BufferedImage that is the blurred version of the given image
+     * This is done by dulling the picture using the convolveOp class
+     */
     private BufferedImage blurImage(BufferedImage image)
     {
-        //TODO code
-        return null;
+        int radius = 11;
+        int size = radius * 2 + 1;
+        float weight = 1.0f / (size * size);
+        float[] data = new float[size * size];
+
+        for (int i = 0; i < data.length; i++) {
+            data[i] = weight;
+        }
+
+        Kernel kernel = new Kernel(size, size, data);
+        ConvolveOp op = new ConvolveOp(kernel, ConvolveOp.EDGE_ZERO_FILL, null);
+        image = op.filter(image, null);
+        return image;
     }
 
+    /**
+     *
+     * @param blurredImage that will be added and divided by the
+     * @param grayImage
+     * @return A BufferedImage that has sharpened the outlines and white
+     * background to set up the final coloring page look that is desired
+     */
     private BufferedImage dodgeAndMerge(BufferedImage blurredImage, BufferedImage grayImage)
     {
-        //TODO code
-        return null;
+        BufferedImage dividedImage = clone(blurredImage);
+        for(int x = 0; x < grayImage.getHeight(); x++)
+        {
+            for (int y = 0; y < grayImage.getWidth(); y++)
+            {
+                dividedImage.setRGB(x, y, (blurredImage.getRGB(x,y)+grayImage.getRGB(x,y))/grayImage.getRGB(x,y));
```

Comment:
This will throw an error if grayImage.getRGB(x,y) is equal to 0, so you need to check if it's 0 before you divide.
Maybe save the value of grayImage.getRGB(x,y) in a variable and then if it's equal to 0, just add 1 since that will be close enough,